app/v2/controllers/V2DeparturesController.scala

```diff
@@ -159,20 +161,42 @@ class V2DeparturesControllerImpl @Inject() (
         .asPresentation
     } yield declarationResult
 
-  def getMessage(departureId: MovementId, messageId: MessageId): Action[AnyContent] =
+  def getMessage(departureId: MovementId, messageId: MessageId, convertBodyToJson: Boolean): Action[AnyContent] = {
```

Comment:
I don't like this boolean being here, because it's not clear what's going on, and if we for some reason want to add another format this will need to be changed.

Along with moving the selection logic to here, I'd create a sealed trait or abstract class like so:

```scala
sealed trait MessageFormat {
    def convertFromXML(message: String, conversionService: ConversionService): EitherT[Future, ConversionError(?), String] // or whatever
}

object XML {
    def convertFromXML(message: String, conversionService: ConversionService): EitherT[Future, ConversionError(?), String] == ??? // or whatever -- will be the identity
}

object Json {
 ...
}
``` 

and that way we can move some of this code away from here and make it more generic if we need it later.

Reply:
(note that names and such aren't final, more interested in getting the idea across with this!)